packages/reusables/src/components/ui/input-otp.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { Pressable, Text, View, type ViewProps } from 'react-native';
+import Animated, { withRepeat, withSequence, withTiming, useAnimatedStyle } from 'react-native-reanimated';
+import { cn } from '../../lib/utils';
+import { Input } from './input';
+
+interface InputOTPProps extends ViewProps {
+  value?: string;
+  maxLength?: number;
+  onValueChange?: (value: string) => void;
+  pattern?: RegExp;
+  separator?: React.ReactNode;
+  containerClassName?: string;
+  onBlur?: () => void;
+  groupSize?: number; // Size of each group (e.g., 2 for XX-XX-XX format)
+}
+
+const InputOTP = React.forwardRef<React.ElementRef<typeof View>, InputOTPProps>(
+  (
+    {
+      className,
+      containerClassName,
+      value = '',
+      maxLength = 6,
+      pattern = /^[0-9]$/,
+      onValueChange,
+      separator = '-',
+      groupSize = 1,
+      onBlur,
+      ...props
+    },
+    ref
+  ) => {
+    const [focused, setFocused] = React.useState(false);
+    const inputRef = React.useRef<React.ElementRef<typeof Input>>(null);
+
+    const handlePress = () => {
+      inputRef.current?.focus();
+    };
+
+    const handleChangeText = (text: string) => {
+      if (text.length <= maxLength && text.split('').every((char) => pattern.test(char))) {
+        onValueChange?.(text);
+      }
+    };
+
+    const handleBlur = () => {
+      setFocused(false);
+      onBlur?.();
+    };
+
+    // Create groups of digits based on groupSize
+    const renderSlots = () => {
+      const slots = [];
+      for (let i = 0; i < maxLength; i++) {
+        // Add separator between groups
+        if (i > 0 && i % groupSize === 0 && separator) {
+          slots.push(
+            <Text key={`sep-${i}`} className='text-muted-foreground px-0.5'>
+              {separator}
+            </Text>
+          );
+        }
+        
+        slots.push(
+          <InputOTPSlot
+            key={i}
+            char={value[i]}
+            focused={focused && value.length === i}
+            filledCharacters={value.length}
+          />
+        );
+      }
+      return slots;
+    };
+
+    return (
+      <View
+        ref={ref}
+        className={cn('flex flex-row items-center justify-center', containerClassName)}
+        {...props}
+      >
+        <Input
+          ref={inputRef}
+          className='absolute opacity-0 w-0 h-0'
+          maxLength={maxLength}
+          keyboardType='number-pad'
```

Comment:
The `keyboardType` is hard coded. You could offer it as a prop and have the default value of `number-pad`